internal/rootcoord/create_collection_task.go

```diff
@@ -44,9 +44,13 @@ import (
 
 type createCollectionTask struct {
 	*Core
-	Req    *milvuspb.CreateCollectionRequest
-	header *message.CreateCollectionMessageHeader
-	body   *message.CreateCollectionRequest
+	Req             *milvuspb.CreateCollectionRequest
+	header          *message.CreateCollectionMessageHeader
+	body            *message.CreateCollectionRequest
+	preserveFieldID bool
+	// If set, use these pchannels instead of load-balanced allocation.
+	// Used by snapshot restore to preserve pchannel mapping from the source collection.
+	preferredPChannels []string
```

Comment:
is there a reason we must reserve pchannels?

Reply:
for kafka/pulsar/woodpacker, start position is not shared among all channels, so if we preserve a dml position from pchannelA, and new collection map to pchannelB and use this position to seek, it will cause unexpected error